builder/dockerfile/dispatchers.go

```diff
@@ -71,6 +71,10 @@ func env(b *Builder, args []string, attributes map[string]bool, original string)
 			return errBlankCommandNames("ENV")
 		}
 
+		if runtime.GOOS == "windows" {
+			// Case insensitive environment variables on Windows
+			args[j] = strings.ToUpper(args[j])
```

Comment:
I'd say #28770 has the same problem. It should only do the case insensitive comparison and leave the last key in its original form. Also see, https://github.com/docker/docker/blob/master/daemon/commit.go#L46-L56 for more unhandled cases.